tasklist/client/src/v2/TaskDetails/Variables/index.tsx

```diff
@@ -237,7 +240,19 @@ const Variables: React.FC<Props> = ({
                           containerRef={formRef}
                           variables={variables}
                           readOnly={!canCompleteTask}
-                          fetchFullVariable={fetchFullVariable}
+                          fetchFullVariable={async (variableKey) => {
+                            setVariablesLoadingFullValue((variableKeys) => [
+                              ...variableKeys,
+                              variableKey,
+                            ]);
+                            await fetchFullVariable({
+                              variableKey,
+                              userTaskKey: task.userTaskKey,
+                            });
+                            setVariablesLoadingFullValue((variableKeys) =>
+                              variableKeys.filter((key) => key !== variableKey),
+                            );
+                          }}
```

Comment:
I am thinking that the loading state can be calculated from inside the `VariableEditor` component. this way, we would not need `variablesLoadingFullValue` prop anymore.

Reply:
I only lifted this state because I want to keep the components in sync across both versions and their slightly different implementations.
So for now I'll leave it as is, if you feel strongly about it we can discuss it and I'll open another PR fixing it